mod-audit-server/src/main/java/org/folio/services/marc/impl/MarcAuditServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.services.marc.impl;
+
+import io.vertx.core.Future;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.marc.MarcAuditDao;
+import org.folio.dao.marc.MarcAuditEntity;
+import org.folio.services.marc.MarcAuditService;
+import org.folio.util.marc.ParsedRecordUtil;
+import org.folio.util.marc.SourceRecordDomainEvent;
+import org.springframework.stereotype.Service;
+
+import static org.folio.util.ErrorUtils.handleFailures;
+
+@Service
+public class MarcAuditServiceImpl implements MarcAuditService {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  private final MarcAuditDao marcAuditDao;
+
+  public MarcAuditServiceImpl(MarcAuditDao marcAuditDao) {
+    this.marcAuditDao = marcAuditDao;
+  }
+
+  @Override
+  public Future<RowSet<Row>> saveMarcDomainEvent(SourceRecordDomainEvent event) {
+    var tenantId = event.getEventMetadata().getTenantId();
+    LOGGER.debug("saveOrderAuditEvent:: Trying to save SourceRecordDomainEvent tenantId: '{}', eventId: '{};", tenantId, event.getEventId());
+    MarcAuditEntity entity;
+    try {
+      entity = ParsedRecordUtil.mapToEntity(event);
+    } catch (Exception e) {
+      LOGGER.warn("saveMarcBibDomainEvent:: Error during mapping SourceRecordDomainEvent to MarcAuditEntity for event '{}'", event.getEventId());
+      return handleFailures(e, event.getEventId());
+    }
+    if (entity.diff() == null || entity.diff().isEmpty()) {
```

Comment:
@dmytrokrutii  Do we need there one more condition on verify `entity.recordType() == null`? Because in DAO we have this piece of code: `recordType.equals(SourceRecordType.MARC_BIB)`. This equals can cause NPE.

Reply:
We do not expect this value to be null in the event. 
Anyway, I add @NonNull annotation to fields in MarcAuditEntity and change condition in DAO to `SourceRecordType.MARC_BIB.equals(recordType)`